### ISSUE TITLE
Cannot find device "lxcbbbfc"

### ISSUE THREAD

**Issue by borkmann** [cilium/cilium#633]
Also here seen while 11-getting-started running in a loop:
```
[...]
May 02 13:07:25 linux-3-1.home cilium-agent[28843]: INFO Regenerated program of endpoint 14594
May 02 13:07:26 linux-3-1.home cilium-agent[28843]: INFO Regenerated program of endpoint 28707
May 02 13:07:27 linux-3-1.home cilium-agent[28843]: INFO Regenerated program of endpoint 14594
May 02 13:07:27 linux-3-1.home cilium-agent[28843]: INFO Regenerated program of endpoint 28707
May 02 13:07:28 linux-3-1.home cilium-agent[28843]: INFO Regenerated program of endpoint 14594
May 02 13:07:28 linux-3-1.home cilium-agent[28843]: INFO Regenerated program of endpoint 28707
May 02 13:07:29 linux-3-1.home cilium-agent[28843]: INFO New policy imported, regenerating...
May 02 13:07:29 linux-3-1.home cilium-agent[28843]: WARN Command execution failed: exit status 1
May 02 13:07:29 linux-3-1.home cilium-agent[28843]: WARN Command output:
May 02 13:07:29 linux-3-1.home cilium-agent[28843]: Join EP id=28707_next ifname=lxcbbbfc
May 02 13:07:29 linux-3-1.home cilium-agent[28843]: Cannot find device "lxcbbbfc"
May 02 13:07:29 linux-3-1.home cilium-agent[28843]: Cannot find device "lxcbbbfc"
May 02 13:07:30 linux-3-1.home cilium-agent[28843]: INFO Regenerated program of endpoint 14594
May 02 13:07:31 linux-3-1.home cilium-agent[28843]: INFO Regenerated program of endpoint 56851
May 02 13:07:33 linux-3-1.home cilium-agent[28843]: INFO Regenerated program of endpoint 14594
May 02 13:07:34 linux-3-1.home cilium-agent[28843]: INFO Regenerated program of endpoint 56851
May 02 13:07:35 linux-3-1.home cilium-agent[28843]: INFO Regenerated program of endpoint 14594
[...]
```

**Comment by aanm** [comment]
It seems that the container was removed by the time the `Join EP id=28707_next ifname=lxcbbbfc` was called.

**Comment by tgraf** [comment]
Is this a race in the build queue where it does not notice that the endpoint has been gone already?

**Comment by aanm** [comment]
Not the build queue itself, the container can be removed while the program is being build in this line:
https://github.com/cilium/cilium/blob/master/pkg/endpoint/bpf.go#L228
After `clang` is done, the container and interface are gone, `tc` is called and doesn't find the interface throwing that error.

**Comment by tgraf** [comment]
Fixed by #695
